frontend-react/src/components/Admin/AdminModal.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+    Button,
+    ButtonGroup,
+    Modal,
+    ModalFooter,
+    ModalHeading,
+    ModalRef,
+    ModalToggleButton,
+} from "@trussworks/react-uswds";
+import React, { RefObject } from "react";
+import { ButtonProps } from "@trussworks/react-uswds/lib/components/Button/Button";
+
+interface ModalConfirmButtonProps {
+    uniquid: string;
+    handleClose: () => void;
+}
+
+export const ModalConfirmDeleteButton = ({
+    handleClose,
+    uniquid,
+    ...buttonProps
+}: ModalConfirmButtonProps &
+    Omit<ButtonProps, "type"> &
+    JSX.IntrinsicElements["button"]): React.ReactElement => {
+    return (
+        <Button
+            {...buttonProps}
+            aria-label="Confirm deleting this item"
+            onClick={handleClose}
+            data-close-modal
+            key={`${uniquid}-confirm-button`}
+            data-uniqueid={uniquid}
+            type="button"
+        />
+    );
+};
+
+ModalConfirmDeleteButton.displayName = "ModalConfirmDeleteButton";
+
+interface DeleteSettingsModalProps {
+    uniquid: string;
+    onConfirm: () => void;
+    modalRef: RefObject<ModalRef>;
+}
+
+export const ConfirmDeleteSettingModal: React.FC<DeleteSettingsModalProps> = ({
+    uniquid,
+    onConfirm,
+    modalRef,
+}) => {
+    const scopedConfirm = () => {
+        modalRef?.current?.toggleModal(undefined, false);
+        onConfirm();
+    };
+
+    return (
+        <>
+            <Modal
+                ref={modalRef}
+                id={uniquid}
+                aria-labelledby={`${uniquid}-heading`}
+                aria-describedby={`${uniquid}-description`}
+            >
+                <ModalHeading id={`${uniquid}-heading`}>
+                    Are you sure you want to continue?
+                </ModalHeading>
+                <div className="usa-prose">
+                    <p id={`${uniquid}-description`}>
+                        You are about to delete this setting.
```

Comment:
nice to have: "You are about to delete ${name}" just in case they clicked a neighbor delete button

Reply:
@sean-usds agreed, fixed. @TomNUSDS I seem to be getting now a warning about missing attribute 'children' on the ModalConfirmDeleteButton, but couldn't find the magic sauce to satisfy it. 
![image](https://user-images.githubusercontent.com/92405130/150436925-2c27777f-3982-40b8-8ee3-864896579c5d.png)

Reply:
"children" means stuff inside the `<Button>` ... so `<Button ...>_children_</Button>`

Reply:
Anyone playing along, the solution was to use `<Button...>{buttonProps.children}</Button>`